bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/RuleManager.java

```diff
@@ -0,0 +1,248 @@
+ */
+public abstract class RuleManager {
+
+    public static final String ID_PREFIX = "rule_"; //$NON-NLS-1$
+    private static RuleEngine re;
+
+    private static int maxId;
+
+    public RuleManager(BundleContext bc) {
+        Set<RuleImpl> rules = loadRules();
+        if (rules == null) {
+            rules = new HashSet<RuleImpl>(20);
+        }
+        maxId = getMaxId(ID_PREFIX, rules) + 1;
+        if (re == null) {
+            re = new RuleEngine(bc);
+        }
+        for (Iterator<RuleImpl> it = rules.iterator(); it.hasNext();) {
+            re.setRule(it.next());
+        }
+    }
+
+    public void dispose() {
+        if (re != null) {
+            re.dispose();
+            re = null;
+        }
+    }
+
+    /**
+     *
+     * @see org.eclipse.smarthome.automation.RuleRegistry#getRule(java.lang.String)
+     */
+    public synchronized Rule getRule(String ruleUID) {
+        RuleImpl oldR = re.getRule(ruleUID);
+        if (oldR == null)
+            return null;
+        return new RuleImpl(oldR);
+    }
+
+    /**
+     * @see org.eclipse.smarthome.automation.RuleRegistry#getRules(java.lang.String)
+     */
+    public synchronized Collection<Rule> getRules(String ruleFilter) {
+        // TODO impl filtering
+        return re.getRules(ruleFilter);
+    }
+
+    /**
+     *
+     * @see org.eclipse.smarthome.automation.RuleRegistry#getRulesByTag(java.lang.String )
+     */
+    public synchronized Collection<Rule> getRulesByTag(String tag) {
+        return re.getRulesByTag(tag);
+    }
+
+    /**
+     *
+     * @see org.eclipse.smarthome.automation.RuleRegistry#addRule(com.prosyst.mbs.services .automation.Rule)
+     */
+    public synchronized void addRule(Rule rule) {
+        addRule0(rule, getScopeIdentifier());
+    }
+
+    /**
+     *
+     * @see org.eclipse.smarthome.automation.RuleRegistry#addRule(com.prosyst.mbs.services .automation.Rule,
+     *      java.lang.String)
+     */
+    public synchronized void addRule(Rule rule, String identity) {
+        // TODO check permissions
+        addRule0(rule, identity);
+    }
+
+    /**
+     * @param rule
+     * @param identity
+     * @throws IllegalArgumentException when the rule is already added or tring to add illegal instance of rule.
+     */
+    private void addRule0(Rule rule, String identity) {
+        if (!(rule instanceof RuleImpl)) {
+            throw new IllegalArgumentException("Illegal instance of Rule: " + rule);
+        }
+        RuleImpl r = (RuleImpl) rule;
+        String rUID = r.getUID();
+        if (rUID != null) {
+            if (re.getRule(rUID) != null) {
+                throw new IllegalArgumentException("The rule: " + rUID + " is already added.");
+            }
+        } else {
+            rUID = getUniqueId();
+            r.setUID(rUID);
+        }
+        RuleImpl r1 = new RuleImpl(r);
+        r1.setScopeIdentifier(identity);
+        r1.setUID(rUID);
+        storeRule(r1);
```

Comment:
This must somehow be delegated back to the RuleRegistryImpl through an ```update(r1)```, since this will take care of the persistence.